### PR TITLE
feat(/xref/filter): add UI to filter raw xref data

### PR DESCRIPTION
**Pull request by sidvishnoi**
Demo: https://i.imgur.com/8hPzaJj.mp4

### CODE REVIEW

File: static/xref/filter/index.html [hunk 1]
```diff
@@ -0,0 +1,64 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title>Search cross references</title>
+    <link rel="stylesheet" href="style.css?v=2020-06-03" />
```
Comment (marcoscaceres) [inline static/xref/filter/index.html:7]: For shame!... but ok :) we could tell it to revalidate via HTTP, no?

File: static/xref/filter/worker.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
+let data = [];
```
Comment (marcoscaceres) [inline static/xref/filter/worker.js:1]: why "global" data?

File: static/xref/filter/worker.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  const query = rawQuery
+    .split('\n')
+    .filter(s => s.trim() && !s.startsWith('//'))
+    .map(s => `(${s.trim().replace(singleEqualSign, '===')})`)
+    .join(' && ');
+  const body = `
+    const { term, uri, spec, shortname, type, for: forContext, normative, status } = entry;
+    return (${query || true});
+  `;
+  console.debug(body);
```
Comment (marcoscaceres) [inline static/xref/filter/worker.js:41]: stray console, or you want this one?

Comment (sidvishnoi) [reply static/xref/filter/worker.js:41]: Want this one.

File: static/xref/filter/worker.js (hunk 2 above)
Comment (sidvishnoi) [reply static/xref/filter/worker.js:1]: We fetch once on start, then reuse that data on all further queries

File: static/xref/filter/index.html (hunk 1 above)
Comment (sidvishnoi) [reply static/xref/filter/index.html:7]: I was thinking of using immutable cache on static asset files through server.

Comment (marcoscaceres) [reply static/xref/filter/index.html:7]: yeah, that could be nice.

**Pull request merged**